### PR TITLE
Optimized string concatenation in `_Tzdb_update_version()`

### PR DESCRIPTION
**Pull request by Mq-b**

### CODE REVIEW

**Review by frederick-vs-ja — CHANGES_REQUESTED** [review]
I'm not sure whether there's currently any optimization except for not calculating the length of `"."`.
Note that the result object is now copied from the concatenated string, because `operator+=` returns an lvalue, so the current change is possibly a **pessimization**.

Perhaps we can optimize this function by ensuring that there's only one allocation - the length of the resulted string can be easily calcluated before any allocation.

**Comment by Mq-b** [comment]
```cpp
std::string s;
s.resize(33);
s = "😅";
std::cout << (const void*)s.data() << '\n';
auto s2 = std::move(s) + "🤣";
std::cout << (const void*)s2.data() << '\n';
```

`std::move` is not meaningless and allows for memory reuse. 

However, changing `operator+` to `operato+=` is fine by me.

**Comment by Mq-b** [comment]
> 
> ```c++
> (_Icu_version += '.') += _STD to_string(_Num_leap_seconds);
> return _Icu_version;
> ```
> 

this is right.

File: stl/inc/chrono [hunk 1]
```diff
@@ -2176,7 +2176,8 @@ namespace chrono {
 
     _NODISCARD inline string _Tzdb_update_version(const string_view _Version, const size_t _Num_leap_seconds) {
         string _Icu_version{_Version.substr(0, _Version.find_last_of('.'))};
-        return _STD move(_Icu_version) + "." + _STD to_string(_Num_leap_seconds);
+        (_Icu_version += '.') += _STD to_string(_Num_leap_seconds);
+        return _Icu_version;
```
Comment (frederick-vs-ja) [inline stl/inc/chrono:2180]: If I understand correctly, the new version dynamically allocates exactly the same time(s) as the existing one. Could you explain in the description of this pull request what is considered optimized?

----
Given that:
- changing from `"."` to `'.'` makes a simpler overload selected, and
- resuing of `_Icu_version` can reduce the number of `string` object to create,

perhaps we can say this PR can save throughput, and improve performance in debug mode.

File: stl/inc/chrono [hunk 2]
```diff
@@ -2176,7 +2176,8 @@ namespace chrono {
 
     _NODISCARD inline string _Tzdb_update_version(const string_view _Version, const size_t _Num_leap_seconds) {
         string _Icu_version{_Version.substr(0, _Version.find_last_of('.'))};
-        return _STD move(_Icu_version) + "." + _STD to_string(_Num_leap_seconds);
+        (_Icu_version += '.') += _STD to_string(_Num_leap_seconds);
```
Comment (StephanTLavavej) [inline stl/inc/chrono:2179]: We conventionally avoid complicated subexpressions (especially in newer code). Spending editor lines to improve clarity is a worthwhile cost. Here, we should perform the `+=` concatenations on separate lines.

**Review by StephanTLavavej — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks! I pushed a commit to perform the concatenations on separate lines, since we conventionally don't perform multiple compound assignments on a single line.

Overall, I think this change is reasonable - taking advantage of the NRVO, and using the single-character overload instead of the one for arbitrary C strings. This code is not actually exercised in normal usage (it's bypassed unless the registry reports new leap seconds that we don't know about), so I don't think this is worth extreme micro-optimization, but I'm okay with the change here.

I manually tested this codepath since normal execution doesn't exercise it, and debugged into it to verify that the new code was getting picked up (i.e. verifying that I hadn't forgotten to `set_environment.bat`):

```
C:\GitHub\STL\out\x64>type meow.cpp
```
```cpp
#include <chrono>
#include <print>
using namespace std;
using namespace std::chrono;

int main() {
    const tzdb& db = get_tzdb();
    println("Original version: {}", db.version);
    println(" Updated version: {}", _Tzdb_update_version(db.version, 1729));
}
```
```
C:\GitHub\STL\out\x64>cl /EHsc /nologo /W4 /std:c++latest /MTd /Od /Zi /Fdmeow.pdb meow.cpp && meow
meow.cpp
Original version: 2021a.27
 Updated version: 2021a.1729
```

We merge changes to our GitHub and MSVC-internal repos simultaneously, using a semi-manual "mirroring" process. To save time, we batch up PRs. Your PR will be part of the next batch! I'll post comments here as I prepare that merge.

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Review by frederick-vs-ja — APPROVED** [review]

**Comment by StephanTLavavej** [comment]
Thanks for making the codebase more consistent about using the Named Return Value Optimization, and congratulations on your first microsoft/STL commit! :tada: :heart_eyes_cat: :rocket:

This change is expected to ship in VS 2022 17.11 Preview 2.

**Pull request merged**